db/db_impl/db_impl_write.cc

```diff
@@ -60,6 +66,145 @@ Status DBImpl::WriteWithCallback(const WriteOptions& write_options,
 }
 #endif  // ROCKSDB_LITE
 
+Status DBImpl::MultiThreadWrite(const WriteOptions& options,
+                                const std::vector<WriteBatch*>& updates) {
+  if (immutable_db_options_.enable_multi_thread_write) {
+    if (UNLIKELY(updates.empty())) {
+      return Status::OK();
+    }
+    autovector<WriteBatch*> batches;
+    for (auto w : updates) {
+      batches.push_back(w);
+    }
+    return MultiThreadWriteImpl(options, batches, nullptr, nullptr);
+  } else {
+    Status status;
+    for (auto b : updates) {
+      status = WriteImpl(options, b, nullptr, nullptr);
+      if (!status.ok()) {
+        return status;
+      }
+    }
+    return status;
+  }
+}
+
+Status DBImpl::MultiThreadWriteImpl(const WriteOptions& write_options,
+                                    const autovector<WriteBatch*>& updates,
+                                    WriteCallback* callback, uint64_t* log_used,
+                                    uint64_t log_ref, uint64_t* seq_used) {
+  StopWatch write_sw(env_, immutable_db_options_.statistics.get(), DB_WRITE);
+  WriteThread::Writer writer(write_options, updates, callback, log_ref);
+  write_thread_.JoinBatchGroup(&writer);
+
+  WriteContext write_context;
+  PERF_TIMER_GUARD(write_pre_and_post_process_time);
```

Comment:
move this line to before `JoinBatchGroup`?

Reply:
Do you mean 'PERF_TIMER_GUARD(write_pre_and_post_process_time);'?

Reply:
yes